library/mt_ip.py

```diff
@@ -75,7 +78,11 @@ def main():
   )
 
   params = module.params
-  idempotent_parameter = 'name'
+  if params['parameter'] == 'address-list':
+    params['parameter'] = 'firewall/address-list'
```

Comment:
based on my comment below, I now see where you are overriding this.  This violates POLA (https://en.wikipedia.org/wiki/Principle_of_least_astonishment)  

We should make them enter the paramater "firewall/address-list".